lib/settings/persistence/local_storage_settings_persistence.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+import 'settings_persistence.dart';
+
+class LocalStorageSettingsPersistence extends SettingsPersistence {
+  late final SharedPreferences _sharedPreferences;
+  bool _isInitialized = false;
+
+  Future<void> initialize() async {
+    if (_isInitialized) {
+      return;
+    }
+
+    _isInitialized = true;
+
+    _sharedPreferences = await SharedPreferences.getInstance();
+  }
+
+  @override
+  Future<bool> getMusicOn({required bool defaultValue}) async {
+    return _sharedPreferences.getBool('musicOn') ?? defaultValue;
+  }
+
+  @override
+  Future<bool> getSoundsOn({required bool defaultValue}) async {
+    return _sharedPreferences.getBool('soundsOn') ?? defaultValue;
+  }
```

Comment:
The getters of `SettingsPersistence` could be synchronous, could not they ?

Reply:
Refactored